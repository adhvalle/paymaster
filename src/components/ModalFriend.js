import { useState } from 'react'

function ModalFriend ({ showModal, closeModal, addFriend }) {
  if (!showModal) return false

  const [name, updateName] = useState('')

  const handleChange = event => {
    const newName = event.target.value
    updateName(newName)
  }

  const handleSubmit = event => {
    event.preventDefault()
    if (!name) return false
    addFriend(name)
  }

  return (
    <div
      className='modal'
      data-testid='modal'
      onClick={closeModal}
    >
      <div
        className='modal__inner'
        onClick={e => { e.stopPropagation() }}
      >
        <form
          className='form'
          onSubmit={handleSubmit}
        >
          <div>
            <label className='form__label'>Nombre</label>
            <input
              className='form__input'
              type='text'
              value={name}
              onChange={handleChange}
            />
          </div>
          <button>AÃ±adir</button>
        </form>
      </div>
    </div>
  )
}

export default ModalFriend
