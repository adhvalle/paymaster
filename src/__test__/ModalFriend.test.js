import { render, screen, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ModalFriend from '../components/ModalFriend'

test('It should render the component', () => {
  const addFriend = true
  render(<ModalFriend showModal={addFriend} />)

  const app = screen.getByTestId('modal')
  expect(app).toMatchSnapshot()
})

test('It should render the component', async () => {
  const addFriendFn = jest.fn()
  const addFriend = true
  render(<ModalFriend showModal={addFriend} addFriend={addFriendFn} />)

  const input = screen.getByRole('textbox')
  const button = screen.getByRole('button')

  act(() => {
    /* fire events that update state */
    userEvent.type(input, 'new name')
    userEvent.click(button)
  })
  expect(addFriendFn).toHaveBeenCalledWith('new name')
})
