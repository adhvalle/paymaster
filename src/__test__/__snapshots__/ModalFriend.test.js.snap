// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the component 1`] = `
<div
  class="modal"
  data-testid="modal"
>
  <div
    class="modal__inner"
  >
    <form
      class="form"
    >
      <div>
        <label
          class="form__label"
        >
          Nombre
        </label>
        <input
          class="form__input"
          type="text"
          value=""
        />
      </div>
      <button>
        AÃ±adir
      </button>
    </form>
  </div>
</div>
`;
